_rules/widget-non-empty-accessible-name-rdzs6q.md

```diff
@@ -0,0 +1,382 @@
+- do not have a [required context](https://www.w3.org/TR/wai-aria/#scope) role that itself inherits from one of those roles.
+
+Elements with the `option` role are not tested in this rule because they do not meet the definition of a [User interface component](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components). If these elements are presented as user interface components, these need to be tested separately from this rule.
+
+Unlike the [Image button has non-empty accessible name](https://act-rules.github.io/rules/59796f) rule, this rule does not check whether an element's accessible name is different from the default name. While that rule assumes that, for image buttons, the default name is never descriptive, this rule relaxes that assumption to ensure the simplicity of this rule.
+
+### Related rules
+
+- [Image button has non-empty accessible name](https://act-rules.github.io/rules/59796f)
+- [Link has non-empty accessible name](https://act-rules.github.io/rules/c487ae)
+
+### Bibliography
+
+- [Understanding Success Criterion 4.1.2: Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value)
+- [H91: Using HTML form controls and links](https://www.w3.org/WAI/WCAG21/Techniques/html/H91)
+- [H44: Using label elements to associate text labels with form controls](https://www.w3.org/WAI/WCAG21/Techniques/html/H44)
+- [H65: Using the title attribute to identify form controls when the label element cannot be used](https://www.w3.org/WAI/WCAG21/Techniques/html/H65)
+- [ARIA14: Using aria-label to provide an invisible label where a visible label cannot be used](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA14)
+- [ARIA16: Using aria-labelledby to provide a name for user interface controls](https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA16)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+This `input` element has an [accessible name][] because of its [programmatic label](#programmatic-label).
+
+```html
+<label>
+	first name
+	<input />
+</label>
+```
+
+#### Passed Example 2
+
+This `input` element has an [accessible name][] because of its `aria-label` attribute.
+
+```html
+<div>last name</div>
+<input aria-label="last name" disabled />
+```
+
+#### Passed Example 3
+
+This `select` element has an [accessible name][] because of its [programmatic label](#programmatic-label).
+
+```html
+<label for="country">Country</label>
+<select id="country">
+	<option>England</option>
+	<option>Scotland</option>
+	<option>Wales</option>
+	<option>Northern Ireland</option>
+</select>
+```
+
+#### Passed Example 4
+
+This `textarea` element has an [accessible name][] because of its `aria-labelledby` attribute.
+
+```html
+<div id="country">Country</div>
+<textarea aria-labelledby="country"></textarea>
+```
+
+#### Passed Example 5
+
+This `input` element has an [accessible name][] because of its `placeholder` attribute.
+
```

Comment:
Can we keep the note from https://act-rules.github.io/rules/e086e5#passed-example-5?

Note: While the placeholder attribute is sufficient to provide an [accessible name](https://act-rules.github.io/rules/e086e5#accessible-name), a [visible](https://act-rules.github.io/rules/e086e5#visible) [label](https://www.w3.org/TR/WCAG21/#dfn-labels) that does not disappear when a user starts to enter data is still required for [success criterion 3.3.2 Labels or Instructions](https://www.w3.org/TR/WCAG21/#labels-or-instructions).